Support final version of Scala 3

The following tsk script:

```scala
// 2> /dev/null \
/*
scala_version="3.0.0"
source $(curl -sL git.io/boot-tsk | sh)
run
exit
*/

@main def main() = println("ohi")
```

fails to run as it resolves compiler artifact name to `scala3-compiler_3.0.0`, when in fact it is `scala3-compiler_3`:

```
~ ─╼ sh Scala3.scala                                                                                                                                                                                                                                         
Fetching all libraries the script depends upon
Resolution error: Error downloading org.scala-lang:scala3-compiler_3.0.0:latest.release
  not found: /Users/martynas/.ivy2/local/org.scala-lang/scala3-compiler_3.0.0
  not found: https://repo1.maven.org/maven2/org/scala-lang/scala3-compiler_3.0.0/maven-metadata.xml
Error downloading org.scala-lang:scala3-library_3.0.0:latest.release
  not found: /Users/martynas/.ivy2/local/org.scala-lang/scala3-library_3.0.0
  not found: https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3.0.0/maven-metadata.xml
Compiling Scala3 (1 Scala source)
Compiled Scala3 (378ms)
Resolution error: Error downloading org.scala-lang:scala3-library_3.0.0:latest.release
  not found: /Users/martynas/.ivy2/local/org.scala-lang/scala3-library_3.0.0
  not found: https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3.0.0/maven-metadata.xml
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.NoClassDefFoundError: scala/util/CommandLineParser$ParseError
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
	at java.lang.Class.getMethod0(Class.java:3018)
	at java.lang.Class.getMethod(Class.java:1784)
	at sun.launcher.LauncherHelper.validateMainClass(LauncherHelper.java:650)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:632)
Caused by: java.lang.ClassNotFoundException: scala.util.CommandLineParser$ParseError
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 7 more
```

--- a/tsk
+++ b/tsk
@@ -62,7 +62,20 @@ graalvm_version="${graalvm_version:-21.0.0}"
 # ---------------------------------------
 tsk_log="${tsk_log:-${TMPDIR:-"/tmp"}/tsk-$$.log}"
 
-scala_major_minor="$(echo "${scala_version}" | sed 's/\([0-9]*\.[0-9]*\)\..*/\1/g')"
+case "${scala_version}" in
+  *RC*)
+    scala_binary_version="${scala_version}"
+    ;;
+  *M*)
+    scala_binary_version="${scala_version}"
+    ;;
+  2*)
+    scala_binary_version="$(echo "${scala_version}" | sed 's/\([0-9]*\.[0-9]*\)\..*/\1/g')"
+    ;;
+  3*)
+    scala_binary_version="$(echo "${scala_version}" | sed 's/\([0-9]*\)\..*/\1/g')"
+    ;;
+esac
 
 fatal_error() {
   >&2 echo "FATAL ERROR: $*"
@@ -205,8 +218,8 @@ case "${scala_version}" in
     scala_library="org.scala-lang:scala-library:${scala_version}"
     ;;
   3*)
-    scala_compiler="org.scala-lang:scala3-compiler_${scala_version}:latest.release"
-    scala_library="org.scala-lang:scala3-library_${scala_version}:latest.release"
+    scala_compiler="org.scala-lang:scala3-compiler_${scala_binary_version}:latest.release"
+    scala_library="org.scala-lang:scala3-library_${scala_binary_version}:latest.release"
     ;;
   *)
     fatal_error "Sorry, Scala version [${scala_version}] is not supported by TSK"
@@ -335,7 +348,7 @@ bloop_config_template() {
             $(cs_fetch --default ${dependencies} ${scala_library} | as_string_array_elements)
         ],
         "out": "${abs_script_dir}/.bloop/${module}",
-        "classesDir": "${abs_script_dir}/.bloop/${module}/scala-${scala_major_minor}/classes",
+        "classesDir": "${abs_script_dir}/.bloop/${module}/scala-${scala_binary_version}/classes",
         "resources": [],
         "scala": {
             "organization": "org.scala-lang",
@@ -347,7 +360,7 @@ bloop_config_template() {
             "jars": [
                 $(cs_fetch ${compiler_deps} | as_string_array_elements)
             ],
-            "analysis": "${abs_script_dir}/.bloop/${module}/inc_compile_${scala_major_minor}.zip",
+            "analysis": "${abs_script_dir}/.bloop/${module}/inc_compile_${scala_binary_version}.zip",
             "setup": {
                 "order": "mixed",
                 "addLibraryToBootClasspath": true,

